From error27@gmail.com  Thu Apr 22 16:38:42 2010
From: Dan Carpenter <error27@gmail.com>
Date: Tue, 16 Mar 2010 13:03:49 +0300
Subject: Staging: iio: test for failed allocation
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: devel@driverdev.osuosl.org, kernel-janitors@vger.kernel.org, Jonathan Cameron <jic23@cam.ac.uk>, linux-kernel@vger.kernel.org
Message-ID: <20100316100349.GA4219@bicker>
Content-Disposition: inline


We should return test to see if iio_allocate_trigger() fails and return -ENOMEM.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Acked-by: Jonathan Cameron <jic23@cam.ac.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/iio/accel/lis3l02dq_ring.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/staging/iio/accel/lis3l02dq_ring.c
+++ b/drivers/staging/iio/accel/lis3l02dq_ring.c
@@ -493,6 +493,9 @@ int lis3l02dq_probe_trigger(struct iio_d
 	struct lis3l02dq_state *state = indio_dev->dev_data;
 
 	state->trig = iio_allocate_trigger();
+	if (!state->trig)
+		return -ENOMEM;
+
 	state->trig->name = kmalloc(IIO_TRIGGER_NAME_LENGTH, GFP_KERNEL);
 	if (!state->trig->name) {
 		ret = -ENOMEM;
